contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--5.2.0--5.3.0.sql

```diff
@@ -208,6 +208,262 @@ RETURNS INTEGER AS
+		WHEN 4 THEN precision = 13;
+		WHEN 5 THEN precision = 14;
+		WHEN 6 THEN precision = 15;
+		-- typemod = 7 is not possible for time in Babelfish but
+		-- adding the case just in case we support it in future
+		WHEN 7 THEN precision = 16;
+		END CASE;
+	ELSE precision = 0;
+	END CASE;
+	RETURN precision;
+END;
+$$ LANGUAGE plpgsql IMMUTABLE STRICT;
+
+
+CREATE OR REPLACE FUNCTION sys.tsql_type_scale_helper(IN type TEXT, IN typemod INT, IN return_null_for_rest bool) RETURNS sys.TINYINT
+AS $$
+DECLARE
+	scale INT;
+	v_type TEXT COLLATE sys.database_default := type;
+BEGIN
+	IF v_type IS NULL THEN 
+		RETURN -1;
+	END IF;
+
+	IF typemod = -1 THEN
+		CASE v_type
+		WHEN 'date' THEN scale = 0;
+		WHEN 'datetime' THEN scale = 3;
+		WHEN 'smalldatetime' THEN scale = 0;
+		WHEN 'datetime2' THEN scale = 6;
+		WHEN 'datetimeoffset' THEN scale = 6;
+		WHEN 'decimal' THEN scale = 38;
+		WHEN 'numeric' THEN scale = 38;
+		WHEN 'money' THEN scale = 4;
+		WHEN 'smallmoney' THEN scale = 4;
+		WHEN 'time' THEN scale = 6;
+		WHEN 'tinyint' THEN scale = 0;
+		ELSE
+			IF return_null_for_rest
+				THEN scale = NULL;
+			ELSE scale = 0;
+			END IF;
+		END CASE;
+		RETURN scale;
+	END IF;
+
+	CASE v_type 
+	WHEN 'decimal' THEN scale = (typemod - 4) & 65535;
+	WHEN 'numeric' THEN scale = (typemod - 4) & 65535;
+	WHEN 'money' THEN scale = 4;
+	WHEN 'smallmoney' THEN scale = 4;
+	WHEN 'smalldatetime' THEN scale = 0;
+	WHEN 'datetime2' THEN
+		CASE typemod 
+		WHEN 0 THEN scale = 0;
+		WHEN 1 THEN scale = 1;
+		WHEN 2 THEN scale = 2;
+		WHEN 3 THEN scale = 3;
+		WHEN 4 THEN scale = 4;
+		WHEN 5 THEN scale = 5;
+		WHEN 6 THEN scale = 6;
+		-- typemod = 7 is not possible for datetime2 in Babelfish but
+		-- adding the case just in case we support it in future
+		WHEN 7 THEN scale = 7;
+		END CASE;
+	WHEN 'datetimeoffset' THEN
+		CASE typemod
+		WHEN 0 THEN scale = 0;
+		WHEN 1 THEN scale = 1;
+		WHEN 2 THEN scale = 2;
+		WHEN 3 THEN scale = 3;
+		WHEN 4 THEN scale = 4;
+		WHEN 5 THEN scale = 5;
+		WHEN 6 THEN scale = 6;
+		-- typemod = 7 is not possible for datetimeoffset in Babelfish
+		-- but adding the case just in case we support it in future
+		WHEN 7 THEN scale = 7;
+		END CASE;
+	WHEN 'time' THEN
+		CASE typemod
+		WHEN 0 THEN scale = 0;
+		WHEN 1 THEN scale = 1;
+		WHEN 2 THEN scale = 2;
+		WHEN 3 THEN scale = 3;
+		WHEN 4 THEN scale = 4;
+		WHEN 5 THEN scale = 5;
+		WHEN 6 THEN scale = 6;
+		-- typemod = 7 is not possible for time in Babelfish but
+		-- adding the case just in case we support it in future
+		WHEN 7 THEN scale = 7;
+		END CASE;
+	ELSE
+		IF return_null_for_rest
+			THEN scale = NULL;
+		ELSE scale = 0;
+		END IF;
+	END CASE;
+	RETURN scale;
+END;
+$$ LANGUAGE plpgsql IMMUTABLE STRICT;
+
+/*
+ * Updates typmod values in pg_proc for smallmoney/money data types in
+ * PLTSQL procedures/functions, defined in babelfish_namespace_ext schemas.
+ * Sets money typmod to 1245192(19,4) and smallmoney to 655368(10,4) where typmod is -1.
+ */
```

Comment:
Is there any risk for performance? Lets think if an existing customer is mvu'ing to 5.3.0 and have a lot of tables/procedures -- for tables we are scanning entire pg_attribute and pg_class and filtering for our use case. Will the mvu take more time for this? Does mvu have a time out -- I am guessing it should be same as statement timeout since we will run alter extension upgrade.